python/src/aiconfig/editor/server/server.py

```diff
@@ -255,32 +258,40 @@ async def run():
     try:
         result = await state.aiconfig.run(prompt_name, options=inference_options)  # type: ignore
         LOGGER.debug(f"Result: {result=}")
-        result_text = str(
-            state.aiconfig.get_output_text(prompt_name)  # type: ignore
-            #
-            if isinstance(result, list)
-            #
-            else result.data[0]  # type: ignore
-        )
-        return {"message": "Done", "output": result_text}, 200
+        return HttpPostResponse(
+            message="Done",
+            aiconfig=state.aiconfig,
+        ).to_flask_format()
+        # return {"message": "Done", "output": result_text}, 200
     except Exception as e:
         err: Err[str] = core_utils.ErrWithTraceback(e)
         LOGGER.error(f"Failed to run: {err}")
-        return {"message": f"<p>Failed to run: {err}"}, 400
+        return HttpPostResponse(
+            message=f"Failed to run: {err}",
+            code=400,
```

Comment:
nit, could technically be other codes like 500

Reply:
Can you please write the API you'd like here: https://docs.google.com/document/d/1YwUlIQZX8CiuvtbVR4OvpFQ2nphFITRD_XZR3CDNtoE/edit#bookmark=id.q4291y5qw2us

Input/output shapes, http code semantics